Added test for creating record

AnastasiaShemyakinskaya opened the pull request.

AnastasiaShemyakinskaya committed:
Added test for creating record

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com

AnastasiaShemyakinskaya committed:
Removed comments

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/CreateRecordTest.java:
```diff
@@ -0,0 +1,63 @@
+    try {
+      mockProducer.initTransactions();
+      mockProducer.beginTransaction();
+      mockProducer.send(
```
this call returns the result which should be tested accordingly

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/CreateRecordTest.java:
```diff
@@ -0,0 +1,63 @@
+    final ProducerRecord<String, String> producerRecord =
+        new ProducerRecord<>("topic", "key", data);
+    try {
+      mockProducer.initTransactions();
```
is it possible to write the record without the transactions usage?

AnastasiaShemyakinskaya replied:
Sure, I used transactions because I misunderstood the documentation. Now I see that it's pointless to use them in this code.

AnastasiaShemyakinskaya committed:
Addded tests for RecordMetadata and cleaned the code
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com
